packages/e2e/test/k6/scenarios/idle-clients.js

```diff
@@ -0,0 +1,220 @@
+      startAt += pageSize;
+      txCount = pageResults.length;
+    } while (txCount === pageSize);
+  }
+};
+
+/**
+ * Grab the wallets json file to be used by the scenario.
+ * Group the addresses per wallet (single address or HD wallets).
+ */
+export function setup() {
+  console.log(
+    `Ramp-up: ${RAMP_UP_DURATION}; Sustain: ${STEADY_STATE_DURATION}; Poll: ${POLL_INTERVAL}s; Block height change every: ${NUM_TIP_QUERIES}`
+  );
+  const res = http.get(WALLET_ADDRESSES_URL);
+  check(res, { 'get wallets': (r) => r.status >= 200 && r.status < 300 });
+
+  const { body: resBodyWallets } = res;
+  const walletsOrig = JSON.parse(resBodyWallets);
+  const walletsOrigCount = walletsOrig ? walletsOrig.length : 0;
+  check(walletsOrigCount, {
+    'At least one wallet is required to run the test': (count) => count > 0
+  });
+  console.log(`Wallet addresses configuration file contains ${walletsOrigCount} addresses`);
+
+  // Remove "big transaction history wallets"
+  const filteredWallets = walletsOrig.filter(({ tx_count }) => tx_count < hdWalletParams.maxTxHistory);
+  // Create chunks of `activeAddrCount` addresses per HD wallet
+  const wallets = chunkArray(filteredWallets, hdWalletParams.activeAddrCount);
+
+  const requestedAddrCount = MAX_VU * hdWalletParams.activeAddrCount;
+  const availableAddrCount = filteredWallets.length;
+  if (availableAddrCount < requestedAddrCount) {
+    console.warn(
+      `Requested wallet count * addresses per wallet: (${requestedAddrCount}), is greater than the available addresses: ${availableAddrCount}. Some addresses will be reused`
+    );
+  }
+
+  const tipRes = cardanoHttpPost(TIP_URL);
+  check(tipRes, { 'Initial tip query': (r) => r.status >= 200 && r.status < 300 });
+  const { body } = tipRes;
+  const { blockNo } = JSON.parse(body);
+
+  // When querying transactions, assume the last transaction was done in the previous epoch
+  const blocksPerEpoch = 20000;
+  const blockHeightOfLastTx = blockNo - blocksPerEpoch;
+  check(blockHeightOfLastTx, { 'Block height of last tx (tip - 1 epoch) is valid': (height) => height > 0 });
+
+  return { blockHeightOfLastTx, wallets: wallets.slice(0, MAX_VU) };
+}
+
+/**
+ * Each wallet consisting of hdWalletParams.activeAddrCount addresses is polling the tip, then queries:
+ * - current utxo set for all addresses
+ * - transaction history since last known transaction block height for all addresses
+ */
+// eslint-disable-next-line func-style
+export function scenario_1({ wallets, blockHeightOfLastTx }) {
+  // Get the wallet for the current virtual user
+  // eslint-disable-next-line no-undef
+  const vu = __VU;
+  const wallet = wallets[vu % wallets.length]; // each wallet is a collection of addresses
+  const addresses = wallet.map(({ address }) => address);
+  for (let i = 0; i < NUM_TIP_QUERIES; i++) {
+    cardanoHttpPost(TIP_URL);
+    // No sleep after last query - fetch utxo and tx history immediately
+    if (i < NUM_TIP_QUERIES) {
```

Comment:
Isn't this always `true` as it's checked in the `for`?
```suggestion
    if (i + 1 < NUM_TIP_QUERIES) {
```